arch/cpu/cc26x0-cc13x0/rf-core/prop-mode.c

```diff
@@ -133,7 +133,9 @@ static int off(void);
 static rfc_propRxOutput_t rx_stats;
 /*---------------------------------------------------------------------------*/
 /* Defines and variables related to the .15.4g PHY HDR */
-#define DOT_4G_MAX_FRAME_LEN    2047
+#ifndef DOT_4G_MAX_FRAME_LEN
+  #define DOT_4G_MAX_FRAME_LEN    2047
```

Comment:
Do you feel that changing this constant leads to any difference in performance? In any case, the value 2047 is defined by the standard, it feels wrong to make it user-configurable.

Reply:
Yes, the radio will attempt to receive maximum number of bytes before failing. It's clear that it's not an issue if all neighboring devices are well behaving 802.15.4 devices. There is some other device(s) at the site which sends data with the same PHY layer which triggers this. 

I get your point, i'm not sure if it holds true for all variations of 15.4 spec, in our cases there will never be frames larger than 128bytes. From previous projects we've seen just doubling the frame size and you get a higher of corruption/data loss (using 2047 will be downright impossible with low data rate and dense networks).

I think the practical solution would be to use the maximum supported payload size: `MIN(MAX_PAYLOAD_LEN_WITH_META, 2047)` or making in user configurable.

I'm open for any solution that can make the radio terminate early on packets it knows can't be processed.

Reply:
I'm thinking we should set the value of `maxPktLen` equal to the maximum of actually supported packet size in terms of rx buffer space - basically what you propose. The value 2047 probably should just go in a comment next to the `MAX_PAYLOAD_LEN` constant definition.

Reply:
That works out for me.

Reply:
In the IEEE mode, we do not set this field at all - does this mean unlimited max packet?! This should be fixed then as well.

Reply:
AFAIK the IEEE mode has no parameter to set the maximum supported length.